File: C++ programs/Problems on Recursion/Problems on String/Accept string & count number of small characters.cpp
/* Write a recursive program which accept string from user and count number of small characters.

Input : HElloWOrlD
Output : 5

*/

#include<iostream>
using namespace std;

int SmallI(char *str)
{
	int iCnt=0;
	
	while(*str!='\0')
	{
	    if((*str>='a')&&(*str<='z'))
		{
		    iCnt++;
		}
        str++;		
    }
    return iCnt;	
}

int SmallR(char *str)
{
	static int iCnt=0;
	
	if(*str!='\0')
	{
	    if((*str>='a')&&(*str<='z'))
		{
		    iCnt++;
		}
        str++;
		SmallR(str);
	}		
    return iCnt;	
}
	
int main()
{
	char Arr[20];
	int iRet=0;
	
	cout<<"Enter string\n";
	scanf("%[^'\n']s",Arr);
	
	iRet=SmallI(Arr);
	cout<<"Small characters are: "<<iRet<<"\n";
	
	iRet=SmallR(Arr);
	cout<<"Small characters are: "<<iRet<<"\n";
	
	return 0;
}	

File: Java programs/Problems on String/Accept string & count number of small characters.java
/* Write a java program which accept string from user and count number of Small Characters.

Input : "Marvellous"
Output : 9    

*/

import java.util.*;

class StringDemo
{
    public int CountSmall(String str)
	{
	   int i=0;
       int iSCnt=0;  

	   while(i<str.length())
	   {
		    char ch=str.charAt(i);
	        if((ch>='a')&&(ch<='z'))
		    {
               iSCnt++;
            }
		    i++;
       }
       return iSCnt;
	}
}

class Lba312
{
	public static void main(String arg[])
	{
		int iAns=0;

		Scanner sobj=new Scanner(System.in);

		System.out.println("Enter a String");
		String str=sobj.nextLine();

		StringDemo dobj=new StringDemo();
		
		iAns=dobj.CountSmall(str);

		System.out.println("Count number of Small Characters is: "+iAns);
	}
}  